### ISSUE TITLE
Brave browser integration

### ISSUE THREAD

**Issue by tneo** [keepassxreboot/keepassxc#2414]
The latest version of the Brave browser is based on the [chromium](https://brave.com/new-brave-browser-release-available-for-general-download/) code base. This allows the browser to install Chrome webstore extensions. It would be awesome that the keepassxc browser integration can be checked on in the keepassxc application as there is support for Chrome, Chromium, Firefox and Vivaldi.

**Comment by droidmonkey** [comment]
That's interesting considering Brave's stance on extensions under Muon engine was: NO

Technically it should work out of the box with absolutely no code changes since the Chromium base looks at the registry entry for Chrome to determine the location of the native messaging client.

**Comment by droidmonkey** [comment]
Yup confirmed, works great.

**Comment by tneo** [comment]
I run on Linux and no matter what I do, I don't get the connection to work with Brave. It works for me in Vivaldi.

**Comment by droidmonkey** [comment]
Oh that was an important piece of info...

**Comment by varjolintu** [comment]
@tneo See [this](https://github.com/keepassxreboot/keepassxc-browser/issues/298). Maybe you need to add a symbolic link under `.brave`, if that's the location for configuration files.

**Comment by tneo** [comment]
@varjolintu I'm not sure where to add the symlink. I added in the root for Brave browser config. But there is no change in behaviour.

**Comment by varjolintu** [comment]
@tneo Are you sure about the location for Brave config or native messaging scripts?

**Comment by tneo** [comment]
@varjolintu As far as I know everything resides in ~/.config/BraveSoftware/Brave-Browser/ and there is where I added the symlink.

**Comment by varjolintu** [comment]
And the script is inside nativemessaging path? If it still doesn't work, I'll run some tests.

**Comment by varjolintu** [comment]
I meant the path Brave uses.

**Comment by tneo** [comment]
These are the only entries with NativeMessagingHosts in there:

> access("/home/tneo/.config/BraveSoftware/Brave-Browser", F_OK) = 0
access("/home/tneo/.config/BraveSoftware/Brave-Browser/NativeMessagingHosts", F_OK) = 0
lstat("/home", {st_mode=S_IFDIR|0755, st_size=18, ...}) = 0
lstat("/home/tneo", {st_mode=S_IFDIR|0755, st_size=4096, ...}) = 0
lstat("/home/tneo/.config", {st_mode=S_IFDIR|0700, st_size=8192, ...}) = 0
lstat("/home/tneo/.config/BraveSoftware", {st_mode=S_IFDIR|0700, st_size=27, ...}) = 0
lstat("/home/tneo/.config/BraveSoftware/Brave-Browser", {st_mode=S_IFDIR|0700, st_size=4096, ...}) = 0
lstat("/home/tneo/.config/BraveSoftware/Brave-Browser/NativeMessagingHosts", {st_mode=S_IFDIR|0700, st_size=6, ...}) = 0
access("/etc/opt/chrome/native-messaging-hosts", F_OK) = 0
lstat("/etc", {st_mode=S_IFDIR|0755, st_size=5962, ...}) = 0
lstat("/etc/opt", {st_mode=S_IFDIR|0755, st_size=12, ...}) = 0
lstat("/etc/opt/chrome", {st_mode=S_IFDIR|0755, st_size=44, ...}) = 0
lstat("/etc/opt/chrome/native-messaging-hosts", {st_mode=S_IFDIR|0755, st_size=78, ...}) = 0

**Comment by varjolintu** [comment]
The browser should try to load `org.keepassxc.keepassxc_browser.json` but it seems that is not included in your file operations.

**Comment by lsorba** [comment]
Getting the same problem with Brave on Mac `Version 0.59.35 Chromium: 72.0.3626.81 (Official Build) (64-bit)`.
In the browser's extension page I get `No connected databases found.`
I enabled integration for the broswers Chromium / Google Chrome.
I don't have any `.config/BraveSoftware` or `.config/chromium` folders in my $HOME

**Comment by FWDekker** [comment]
While adding the symlink works for me on Linux, that solution doesn't work on Windows. I tried the following, but neither of them worked for me:

* Renaming the `org.keepassxc.keepassxc_browser_chrome.json` file in `%LOCALAPPDATA%/keepassxc` to `org.keepassxc.keepassxc_browser_brave.json`.
* Adding `org.keepassxc.keepassxc_browser` as a key to `HKEY_CURRENT_USER\Software\Google\Chrome\NativeMessagingHosts` in regedit.

I ended up using the legacy browser support, and that one works fine.

**Comment by droidmonkey** [comment]
This issue is aimed at Linux.

**Comment by MichaelThessel** [comment]
Can confirm this here. I cannot connect the database. I enabled integrations for all browsers just in case. 

Running Brave: Version 0.60.47 Chromium: 72.0.3626.119 (Official Build) unknown (64-bit)
on Linux

**Comment by mjgiarlo** [comment]
@tneo :speech_balloon: 
> As that is not in the directory .config/BraveSoftware/Brave-Browser/NativeMessagingHosts
> That only resides in the .config/chromium/NativeMessagingHosts
> 
> Adding the symlink to `org.keepassxc.keepassxc_browser.json `in `.config/BraveSoftware/Brave-Browser/NativeMessagingHosts` makes integration work.

This worked for me. Thanks, @tneo!

**Comment by therealtakeshi** [comment]
Running Brave Version 0.61.51 Chromium: 73.0.3683.75 (Official Build) (64-bit) on macOS High Sierra.

I tried a few different things, and I was only able to get it working only once I 1) enabled integration support for Chrome and 2) copied that JSON file from Chrome's NativeMessagingHosts directory into Brave's. The same steps but with Chromium or Vivaldi instead of Chrome did not work.

**Comment by wranvaud** [comment]
Can this be reopened? I'm having this issue on ubuntu and creating symlinks is a workaround only

**Comment by varjolintu** [comment]
This issue is still open. We don't provide an official support for Brave browser. Creating symlinks is your only solution for now.

**Comment by mjgiarlo** [comment]
I've just started working on a PR to add Brave integration to KeePassXC: https://github.com/keepassxreboot/keepassxc/pull/2933 I still need to build and test it (on Linux), and I'm happy to have more :eyes: on the PR as it's been years since I touched C++. If anyone is inclined to partner up on this, let me know. I hope to find time later this week to resume this work.

**Comment by ACheshirov** [comment]
The lack of Brave support is the only thing that stops me from switching to KeePassXC.

**Comment by mjgiarlo** [comment]
> The lack of Brave support is the only thing that stops me from switching to KeePassXC.

The workarounds on https://github.com/keepassxreboot/keepassxc/pull/2933 are fairly straightforward to apply if you're super eager to make the jump. Else, hope I'll find more time for this work soon. In the meantime, I think it's ready for another review? @varjolintu

**Comment by mjgiarlo** [comment]
#2933 has now been merged into the 2.4.1 release branch. Once KPXC 2.4.1 is released, we can close this issue.

**Comment by mjgiarlo** [comment]
@droidmonkey :tada: thank you!

**Comment by marlluslustosa** [comment]
> As that is not in the directory .config/BraveSoftware/Brave-Browser/NativeMessagingHosts
> That only resides in the .config/chromium/NativeMessagingHosts
> 
> Adding the symlink to `org.keepassxc.keepassxc_browser.json `in `.config/BraveSoftware/Brave-Browser/NativeMessagingHosts` makes integration work.

This worked for me!

**Comment by rasdehya** [comment]
@tneo Thanks. 

> Adding the symlink to org.keepassxc.keepassxc_browser.json in .config/BraveSoftware/Brave-Browser/NativeMessagingHosts makes integration work.

Still needed in debian buster stable
